eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/threading/PollingThread.java

```diff
@@ -0,0 +1,353 @@
+
+  /**
+   * Creates a new polling thread with the specified thread polling interval. The default polling
+   * interval start delay is used ({@link #DEFAULT_POLLING_INTERVAL_START_DELAY}).
+   *
+   * @param pollingInterval The thread polling interval.
+   * @param pollingIntervalTimeUnit The thread polling interval time unit.
+   * @param connectorSettings The Ewon connector settings.
+   * @since 1.0.0
+   */
+  public PollingThread(
+      long pollingInterval,
+      TimeUnit pollingIntervalTimeUnit,
+      EwonConnectorSettings connectorSettings) {
+    this(
+        pollingInterval,
+        DEFAULT_POLLING_INTERVAL_START_DELAY,
+        pollingIntervalTimeUnit,
+        connectorSettings);
+  }
+
+  /**
+   * Creates a new polling thread with the specified thread polling interval and polling interval
+   * start delay. The start delay is the amount of time to delay before the first poll interval
+   * starts.
+   *
+   * @param pollingInterval The thread polling interval.
+   * @param pollingIntervalStartDelay The thread polling interval start delay.
+   * @param pollingIntervalTimeUnit The thread polling interval time unit.
+   * @param connectorSettings The Ewon connector settings.
+   * @since 1.0.0
+   */
+  public PollingThread(
+      long pollingInterval,
+      long pollingIntervalStartDelay,
+      TimeUnit pollingIntervalTimeUnit,
+      EwonConnectorSettings connectorSettings) {
+    this.pollingInterval = pollingInterval;
+    this.pollingIntervalStartDelay = pollingIntervalStartDelay;
+    this.pollingIntervalTimeUnit = pollingIntervalTimeUnit;
+    this.connectorSettings = connectorSettings;
+    executorService = Executors.newSingleThreadScheduledExecutor();
+  }
+
+  /**
+   * Starts the polling thread using the specified polling interval, and polling interval start
+   * delay.
+   *
+   * @throws IllegalStateException if the polling thread is already running or the polling thread
+   *     executor service has been shutdown.
+   * @since 1.0.0
+   */
+  public void start() {
+    // Check if the executor service is shutdown
+    checkShutdown();
+
+    // Check if the thread is already running
+    if (isRunning()) {
+      throw new IllegalStateException("Polling thread is already running.");
+    }
+
+    // Schedule the thread to run at the specified interval
+    scheduledFuture =
+        executorService.scheduleAtFixedRate(
+            this, pollingIntervalStartDelay, pollingInterval, pollingIntervalTimeUnit);
+  }
+
+  /**
+   * Stops the polling thread using the specified interrupt behavior.
+   *
+   * @param mayInterruptIfRunning true if in-progress polls should be interrupted; otherwise,
+   *     in-progress polls are allowed to complete
+   * @throws IllegalStateException if the polling thread executor service has been shutdown.
+   * @since 1.0.0
+   */
+  public void stop(boolean mayInterruptIfRunning) {
+    // Check if the executor service is shutdown
+    checkShutdown();
```

Comment:
This checks for done, but there is no check for cancelled. I'm not sure if you want to check for this or not. If the thread is already canceled, then canceling it again should not really change the behavior.

Reply:
I'm not sure if you think an additional check is necessary, but calling `cancel()` on a Future that has already been canceled should have the same result as the original `cancel()` call or no effect at all.